avm/res/sql/instance-pool/main.bicep

```diff
@@ -0,0 +1,102 @@
+metadata name = 'SQL Server Instance Pool'
+metadata description = 'This module deploys an Azure SQL Server Instance Pool.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the instance pool.')
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Required. The subnet ID for the instance pool.')
+param subnetId string
```

Comment:
Should be `subnetResourceId` as per spec [Parameters/Variables for Resource IDs](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr22---category-inputs---parametersvariables-for-resource-ids).